cloudant-sync-datastore-core/src/test/java/com/cloudant/sync/datastore/DatastoreImplForceInsertTest.java

```diff
@@ -505,6 +505,30 @@ public void forceInsert_pickWinnerOfConflicts_Simple_2y_last() throws Exception
     }
 
     @Test
+    public void forceInsert_pickWinnerOfConflicts_TwoOfSameRoot_OneDeleted() throws Exception {
+        DocumentRevision rev1a = createDbObject("1-x", bodyOne);
+        DocumentRevision rev1b = createDbObject("1-x", bodyTwo);
+        DocumentRevision rev2 = createDbObjectDeleted("2-x");
+
+        // Insert the same document twice, so we have two identical roots
+        // We have given them different bodies so we can tell them apart
+        datastore.forceInsert(rev1a, "1-x");
+        datastore.forceInsert(rev1b, "1-x");
+
+        // Now force insert the deleted
+        datastore.forceInsert(rev2, "1-x", "2-x");
+
+        // Get the document
+        DocumentRevision doc = datastore.getDocument(OBJECT_ID);
+
+        // We favour non-deleted nodes so the winner should be a 1-x
+        Assert.assertEquals("The document winner should be the expected revision", "1-x", doc
+                .getRevision());
+        Assert.assertEquals("The document winner should have the expected body", bodyTwo.asMap(),
```

Comment:
As per our discussion the ordering is not guaranteed by our query, but in practice appears to always work. I have done the following to make this more robust:
* changed the query to order by sequence so even if there are duplicates the behaviour of where things get grafted will be guaranteed in future: ac570b0
* 70483d1 added a lot more explanation to the test and changed the duplicate insert to not use `forceInsert` so that when `forceInsert` is updated to protect against duplicates the test won't fail. If we update the schema to prevent duplicates then we probably just have to get rid of the test later.